broker-daemon/block-order-worker/index.js

```diff
@@ -557,6 +557,62 @@ class BlockOrderWorker extends EventEmitter {
     return osm
   }
 
+  /**
+   * Place orders for a block order for a given amount, breaking them up based on the maximum order size
+   * @param  {BlockOrder} blockOrder Block Order to place orders on behalf of
+   * @param  {String}     baseAmount Int64 amount, in the base currency's base units, to place orders for
+   * @return {void}
+   */
+  _placeOrders (blockOrder, baseAmount) {
+    // order params
+    const { baseSymbol, counterSymbol, quantumPrice } = blockOrder
+    const baseEngine = this.engines.get(baseSymbol)
+    const counterEngine = this.engines.get(counterSymbol)
+
+    if (!baseEngine) {
+      throw new Error(`No engine available for ${baseSymbol}`)
+    }
+    if (!counterEngine) {
+      throw new Error(`No engine available for ${counterSymbol}`)
+    }
+
+    const baseMaxPayment = baseEngine.maxPaymentSize
+    const counterMaxPayment = counterEngine.maxPaymentSize
+
+    // our max payment size settings have an implied price. We need to compare that to our actual price
+    // to see which max payment size we're going to run up against.
+    const maxPaymentSizeImpliedPrice = Big(counterMaxPayment).div(baseMaxPayment)
+    let maxBaseAmountPerOrder
+
+    // quantum price is the counter/base (both in quantum units)
+    if (Big(quantumPrice).gte(maxPaymentSizeImpliedPrice)) {
+      // counter for the block order is larger than base (relative to their max payment sizes)
+      maxBaseAmountPerOrder = Big(counterMaxPayment).div(quantumPrice).round(0)
+    } else {
+      // base for the block order is larger than counter (relative to their max payment sizes)
+      maxBaseAmountPerOrder = Big(baseMaxPayment)
+    }
+
+    let baseAmountRemaining = Big(baseAmount)
+    let orderCount = 1
```

Comment:
this never gets incremented? Also since _place_orders can be called multiple times for a blockorder, there is the potential to have a bunch of logs that say `Placing order 1` which might not be what you were looking for?

Reply:
Yep I missed the `++`. Fixing now.

`_placeOrders` can be called more than once, but should be called only once per `workBlockOrder`, so I still think this log is useful - you should have conflicting logs at the same moment.